src/sisp.test.js

```diff
@@ -59,12 +59,218 @@ describe('SISP Payment', () => {
+          merchantRespMerchantRef: '52',
+          merchantRespMerchantSession: 'S20200626121231',
+          merchantRespMessageID: 'Zo23nJ2P9C3i3e5Z70K1',
+          merchantRespPan: '************4242',
+          merchantRespPurchaseAmount: '400.00',
+          merchantRespReferenceNumber: '',
+          merchantRespReloadCode: '',
+          merchantRespTid: '00004787',
+          merchantRespTimeStamp: '2020-06-26 12:11:01',
+          messageType: '10',
+          resultFingerPrint: 'zjuDVwa1ZiAXHUW0PY3ulSOFAcQVdufhOtAxE1q8h+gZhyxqupgfagqZrWtIugY0kb4zi2OLA/aoqAouJyvxDA==',
+          resultFingerPrintVersion: '1'
+        };
+
+        const referenceId = '1:3:testForm';
+        const total = 89;
+
+        const sisp = new Sisp({ posID, posAutCode });
+        const requestFormResult = await sisp.generatePaymentRequestForm(referenceId, total, webhookUrl);
+
+        const formExpected = successData.replace(/\s/g, '');
+
+        const result = await sisp.paymentProcessedSuccessfully(paymentSuccessData);
+        const expected = { code: 200, message: 'Payment processing success', data: paymentSuccessData };
+
+        expect(requestFormResult.replace(/\s/g, '')).toEqual(formExpected);
+        expect(result).toEqual(expected);
+      });
+
+      it('should return the correct response if messageType is M', async () => {
+        paymentSuccessData = {
+          languageMessages: 'pt',
+          merchantResp: 'C',
+          merchantRespAdditionalErrorMessage: '',
+          merchantRespCP: '1047',
+          merchantRespClientReceipt: '',
+          merchantRespEntityCode: '',
+          merchantRespErrorCode: '',
+          merchantRespErrorDescription: '',
+          merchantRespErrorDetail: '',
+          merchantRespMerchantRef: '52',
+          merchantRespMerchantSession: 'S20200626121231',
+          merchantRespMessageID: 'Zo23nJ2P9C3i3e5Z70K1',
+          merchantRespPan: '************4242',
+          merchantRespPurchaseAmount: '400.00',
+          merchantRespReferenceNumber: '',
+          merchantRespReloadCode: '',
+          merchantRespTid: '00004787',
+          merchantRespTimeStamp: '2020-06-26 12:11:01',
+          messageType: 'M',
+          resultFingerPrint: '6/AB7qPfq32CeNDFrvi4iU30vc9jPiJiPds9yiptL1hErpQSXVeLWsvgtUVRv7MuvEJaO6pSkkKCR2X8XORYLg==',
+          resultFingerPrintVersion: '1'
+        };
+
+        const referenceId = '1:3:testForm';
+        const total = 89;
+
+        const sisp = new Sisp({ posID, posAutCode });
+        const requestFormResult = await sisp.generatePaymentRequestForm(referenceId, total, webhookUrl);
+
+        const formExpected = successData.replace(/\s/g, '');
+
+        const result = await sisp.paymentProcessedSuccessfully(paymentSuccessData);
+        const expected = { code: 200, message: 'Payment processing success', data: paymentSuccessData };
+
+        expect(requestFormResult.replace(/\s/g, '')).toEqual(formExpected);
+        expect(result).toEqual(expected);
```

Comment:
Maybe is better we add something like this to be more aligned with the description: expect(result.data.messageType).toEqual(expected.data.messageType); then if this fail we will se the message for example: this expected M but got 50. What do you think @FredSoares